### PR TITLE
CLOUDP-75270: Replace atlas.ProjectIPWhitelist with ProjectIPAccessList

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ CLOUDP-75270

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: internal/store/project_ip_access_lists.go [hunk 1]
```diff
@@ -22,25 +22,25 @@ import (
...
 
-type ProjectIPWhitelistDescriber interface {
-	IPWhitelist(string, string) (*atlas.ProjectIPWhitelist, error)
+type ProjectIPAccessListDescriber interface {
+	IPAccessList(string, string) (*atlas.ProjectIPWhitelist, error)
 }
-type ProjectIPWhitelistLister interface {
-	ProjectIPWhitelists(string, *atlas.ListOptions) ([]atlas.ProjectIPWhitelist, error)
+type ProjectIPAccessListLister interface {
+	ProjectIPAccessList(string, *atlas.ListOptions) ([]atlas.ProjectIPWhitelist, error)
```
Comment (gssbzn) [inline internal/store/project_ip_access_lists.go:31]: super small nit, plural
```suggestion
	ProjectIPAccessLists(string, *atlas.ListOptions) ([]atlas.ProjectIPWhitelist, error)
```

**Review by gssbzn — CHANGES_REQUESTED** [review]
one small nit

**Comment by andreaangiolillo** [comment]
This PR is not ready for review

**Review by gssbzn — APPROVED** [review]
LGTM

**Pull request merged**